packages/lab/src/slider/internal/useRangeSliderThumb.ts

```diff
@@ -168,34 +174,38 @@ export const useRangeSliderThumb = ({
       if (distanceToThumb0 > distanceToThumb1) {
         // Move the second thumb
         newValues[1] = newValue;
-        setIsThumbIndex(1);
+        closestThumbIndex = 1;
       } else if (distanceToThumb0 < distanceToThumb1) {
         // Move the first thumb
         newValues[0] = newValue;
-        setIsThumbIndex(0);
+        closestThumbIndex = 0;
       } else {
         // If distances are equal, determine based on the click position
         if (newValue < newValues[0]) {
           // Clicked position is before both thumbs, move the first thumb
           newValues[0] = newValue;
-          setIsThumbIndex(0);
+          closestThumbIndex = 0;
         } else if (newValue > newValues[1]) {
           // Clicked position is after both thumbs, move the second thumb
           newValues[1] = newValue;
-          setIsThumbIndex(1);
+          closestThumbIndex = 1;
         } else {
           // Clicked position is between the thumbs, move the first thumb
           newValues[0] = newValue;
-          setIsThumbIndex(0);
+          closestThumbIndex = 0;
         }
       }
+      setIsThumbIndex(closestThumbIndex);
```

Comment:
NIT... would say "is" naming convention is conventionally associated with booleans.
I'll push up a small rename to improve readability